packages/caliper-core/lib/worker/caliper-worker.js

```diff
@@ -84,55 +85,56 @@ class CaliperWorker {
      * @param {Object} rateController rate controller object
      * @async
      */
-    async runFixedNumber(workloadModule, number, rateController) {
+    async _runFixedNumber(workloadModule, number, rateController) {
         const stats = this.internalTxObserver.getCurrentStatistics();
-        let error = undefined;
-        while (stats.getTotalSubmittedTx() < number && !error) {
+        const errors = [];
+
+        while (stats.getTotalSubmittedTx() < number) {
             await rateController.applyRateControl();
 
-            // If this function calls this.workloadModule.submitTransaction() too quickly, micro task queue will be filled with unexecuted promises,
-            // and I/O task(s) will get no chance to be execute and fall into starvation, for more detail info please visit:
-            // https://snyk.io/blog/nodejs-how-even-quick-async-functions-can-block-the-event-loop-starve-io/
             await this.setImmediatePromise(() => {
                 workloadModule.submitTransaction()
-                    .catch(err => { error = err; });
+                    .catch(err => {
+                        Logger.error(`Error in worker ${this.workerIndex}, round ${this.currentRound}: ${err}`);
+                        errors.push(err);
+                    });
             });
         }
 
-        if (error) {
-            // Already logged, no need to log again
-            throw error;
+        if (errors.length > 0) {
+            throw new Error(`Errors occurred while submitting transactions on worker ${this.workerIndex} for round ${this.currentRound}`);
         }
 
         await CaliperWorker._waitForTxsToFinish(stats);
     }
 
+
     /**
      * Perform test with specified test duration
      * @param {object} workloadModule The user test module.
      * @param {Object} duration duration to run for
      * @param {Object} rateController rate controller object
      * @async
      */
-    async runDuration(workloadModule, duration, rateController) {
+    async _runDuration(workloadModule, duration, rateController) {
         const stats = this.internalTxObserver.getCurrentStatistics();
         let startTime = stats.getRoundStartTime();
-        let error = undefined;
-        while ((Date.now() - startTime) < (duration * 1000) && !error) {
+        const errors = [];
```

Comment:
Sorry my mistake, it looks like the worker terminates it's run at the first error. Apologies I missed that. It means we don't need to make changes around how errors are handled